### PR TITLE
Add saptune support for SLES

### PR DESCRIPTION
**Pull request by mr-stringer**
This PR adds saptune support for SLES based systems in the `sap_hana_preconfigure` and `sap_netweaver_preconfigure` roles.  As discussed, this replaces the current Ansible SAP note implementation with saptune on SLES systems.

The documentation has been updated to reflect these changes.

Assert mode has also been added for SLES for the above roles.

### CODE REVIEW

File: roles/sap_hana_preconfigure/vars/SLES_15.yml [hunk 1]
```diff
@@ -24,3 +24,8 @@ __sap_hana_preconfigure_packages:
 #
 
 __sap_hana_preconfigure_grub_file: /tmp/grub
+
+# Services that conflict with saptune
+__sap_hana_preconfigure_saptune_conflicting_services:
+  - sapconf
+  - tuned
```
Comment (berndfinger) [inline roles/sap_hana_preconfigure/vars/SLES_15.yml:31]: end of file missing.

File: roles/sap_hana_preconfigure/README.md [hunk 2]
```diff
@@ -1,17 +1,17 @@
 # sap_hana_preconfigure Ansible Role
 
 This role installs additional required packages and performs additional configuration steps for installing and running SAP HANA.
-If you want to configure a RHEL system for the installation and later usage of SAP HANA, you have to first run role sap_general_preconfigure
-and then role sap_hana_preconfigure.
+If you want to configure a RHELsystem for the installation and later usage of SAP HANA, you have to first run role sap_general_preconfigure
```
Comment (berndfinger) [inline roles/sap_hana_preconfigure/README.md:4]: RHELsystem -> RHEL system (unchanged from before)

File: roles/sap_hana_preconfigure/README.md [hunk 3]
```diff
@@ -1,17 +1,17 @@
...
+and then role sap_hana_preconfigure.  However, if we wish to run SLES for HANA, you may run only this role.
```
Comment (berndfinger) [inline roles/sap_hana_preconfigure/README.md:5]: @mr-stringer I understand that for SLES, the role sap_general_preconfigure is not needed at all, neither for NetWeaver nor for HANA. Can you think of a good wording for the README.md file in sap_general_preconfigure? Maybe just a single line at the end of the first chapter, just like here?

File: roles/sap_hana_preconfigure/defaults/main.yml [hunk 4]
```diff
@@ -66,7 +66,7 @@ sap_hana_sps: "0"
 sap_hana_preconfigure_packages: "{{ __sap_hana_preconfigure_packages }}"
 #sap_hana_preconfigure_packages: "{{ __sap_hana_preconfigure_packages_min_install }}"
 
-# SAP HANA requires certain minimum package versions to be supported. These mininum levels are listed in SAP Note 2235581.
+# SAP HANA requires certain minimum package versions to be supported. These minium levels are listed in SAP Note 2235581.
```
Comment (rainerleber) [inline roles/sap_hana_preconfigure/defaults/main.yml:69]:
```suggestion
# SAP HANA requires certain minimum package versions to be supported. These minimum levels are listed in SAP Note 2235581.
```

File: roles/sap_hana_preconfigure/vars/SLES_15.yml (hunk 1 above)
Comment (mr-stringer) [reply roles/sap_hana_preconfigure/vars/SLES_15.yml]: Fixed

File: roles/sap_hana_preconfigure/README.md (hunk 2 above)
Comment (mr-stringer) [reply roles/sap_hana_preconfigure/README.md]: Fixed

File: roles/sap_hana_preconfigure/README.md (hunk 3 above)
Comment (mr-stringer) [reply roles/sap_hana_preconfigure/README.md:5]: I have added a line to the sap_general_preconfigure readme.

```
For SLES systems, this role may not be necessary.  The majority of SAP preparation and tuning is covered by `saptune` which is configured in the `sap_hana_preconfigure` and `sap_netweaver_preconfigure` roles.
```

File: roles/sap_hana_preconfigure/defaults/main.yml (hunk 4 above)
Comment (mr-stringer) [reply roles/sap_hana_preconfigure/defaults/main.yml]: Fixed

**Comment by mr-stringer** [comment]
Added new commit fixing highlighted issues.

File: roles/sap_netweaver_preconfigure/README.md [hunk 5]
```diff
@@ -1,21 +1,24 @@
...
 
-Note
+Note 
 ----
-As per SAP notes 2002167 and 2772999, the role will switch to tuned profile sap-netweaver no matter if another tuned profile
+On RHEL, as per SAP notes 2002167 and 2772999, the role will switch to tuned profile sap-netweaver no matter if another tuned profile
 (e.g. virtual-guest) had been active before or not.
 
+On SLES, this role will switch the saptune solution to the one specified by the configuration and will override any previously set solution.
+The default solution is
```
Comment (berndfinger) [inline roles/sap_netweaver_preconfigure/README.md:20]: It appears that some text is missing after "is", to complete this sentence.

**Review by berndfinger — COMMENTED** [review]
lgtm

File: roles/sap_netweaver_preconfigure/README.md (hunk 5 above)
Comment (sean-freeman) [reply roles/sap_netweaver_preconfigure/README.md]: This was updated to...
> The default solution is `NETWEAVER`

**Review by sean-freeman — APPROVED** [review]
lgtm :+1:

**Comment by berndfinger** [comment]
@mr-stringer We have some ansible-lint errors. Please have a look at the results of the GitHub workflows for sap_hana_preconfigure and sap_netweaver_preconfigure. I am going to remove the ansible-lint errors for the defaults/main.yml files in a separate PR.

**Comment by berndfinger** [comment]
I just checked and found that the trailing spaces in defaults/main.yml are part of your PR. So I'll leave it to you to remove them.

**Review by berndfinger — APPROVED** [review]

**Review by rainerleber — APPROVED** [review]
lgtm

**Pull request merged**